### PR TITLE
tests/rng: replace printf_float with fmt/print_float

### PR DESCRIPTION
**Pull request by cladmi**
### Contribution description

When running the test on `arduino-mega2560` printing the float failed
and was printed as ` ?`.

    Calculated  ? bits of entropy from 10000 samples.

Replace using `printf` floating point printing by using `fmt/print_float`.
Now the test succeeds on `arduino-mega2560`.

I used `6` for the precision because that what was printed for `native` in the firmware output.

### Testing procedure

Run `tests/rng` on an `arduino-mega2560` the test should fail in master and work with this PR

Failure in master:

```
2019-02-12 14:00:58,909 - INFO # Calculated  ? bits of entropy from 10000 samples.
> Timeout in expect script at "child.expect(re.compile(r"Calculated 7\.994\d{3} bits of entropy from 10000 samples\."))" (tests/rng/tests/01-run.py:39)
```

Success with this PR.

```
BOARD=arduino-mega2560 make --no-print-directory -C tests/rng/ flash test
2019-02-12 14:03:30,060 - INFO # Running entropy test, with seed 1337 using constant value.
2019-02-12 14:03:30,061 - INFO #
2019-02-12 14:03:30,175 - INFO # Calculated 0.017260 bits of entropy from 10000 samples.
```

### Issues/PRs references

This was found during release tests: https://github.com/RIOT-OS/Release-Specs/issues/98

### CODE REVIEW

**Comment by cladmi** [comment]
@PeterKietzmann I did not notice that it was complaining, it looks not introduced by my PR but I will handle it.

**Comment by miri64** [comment]
I had a few PRs where I had to piggyback some cppcheck warning fixes or suppressions as well. I think the new version spots a few more new errors.

**Comment by cladmi** [comment]
I added the fix for `cppcheck` before my commit and it was rebased on master by `static-test` so pushed force the rebased version.

File: tests/rng/test.c [hunk 1]
```diff
@@ -154,7 +154,7 @@ void test_distributions(uint32_t samples)
 
         /* count bits */
         for (int i = 0; i < 32; i++) {
-            if (value & (1 << i)) {
+            if (value & (1u << i)) {
```
Comment (miri64) [inline tests/rng/test.c]: Shouldn't it be `1LU` to fit 32-bit?

**Comment by cladmi** [comment]
I just tested with `iotlab-m3` after the rebase and it fails… 
I put the `needs squashing` to prevent merging now.

```
7.994708Calculated  bits of entropy from 10000 samples. 
```

If I put a `fflush(stdout);` before the `print_float` it works.
I noticed it is said in the documentation, I will push a fix http://doc.riot-os.org/group__sys__fmt.html

File: tests/rng/test.c (hunk 1 above)
Comment (cladmi) [reply tests/rng/test.c]: Or even better `UINT32_C(1)` for all architectures indeed. I will update.

**Comment by cladmi** [comment]
Compilation fails for me because of `chronos` that does not know `fflush(stdout)` I will look into it tomorrow.

**Comment by miri64** [comment]
> Compilation fails for me because of `chronos` that does not know `fflush(stdout)` I will look into it tomorrow.

why not just use `print_str()` instead of `printf()` around it?

**Comment by cladmi** [comment]
Compilation fails for me because of `chronos` that does not know `fflush(stdout)` I will look into it tomorrow.

> > Compilation fails for me because of `chronos` that does not know `fflush(stdout)` I will look into it tomorrow.
> 
> why not just use `print_str()` instead of `printf()` around it?

The test script is using `printf` in other places and would need to be flushed anyway before using `print_str`.

I just did the same as what is done for the other use of `fflush(stdout)`.


I tested on `wsn430` and it looks like it cannot switch the entropy source. When calling `source 1` it gets stuck. But the issue was also there with master.
Also on `wsn430-v1_3b` printf was not printing float.

So `printf_float` handling does not currently work for these BSPs.

**Comment by PeterKietzmann** [comment]
@cladmi how do we proceed here? Mark it as known issue for the release?

**Comment by cladmi** [comment]
@PeterKietzmann there are always failure for tests when we do a release for non `iotlab-m3` and `samr21-xpro` boards anyway. And for these two there are also known always failures.

**Comment by PeterKietzmann** [comment]
Confirmed the fix on arduino-mega2560 and z1 nodes. My ACK holds. Please squash

**Comment by cladmi** [comment]
I rebased and merged the commit messages. The changes stayed the same.
I do not have my `arduino-mega2560` and `msba2` accessible now for testing.

I tested on `wsn430-v1_3b` the print is now correct and the test still fails with the same error as in master.

**Comment by PeterKietzmann** [comment]
Re-tested on iotlab-m3, telosb and arduino-mega2560. All good – ACK and go!

**Comment by cladmi** [comment]
Thank you for the review.

**Pull request merged**